canonical-websites/snapcraft.io
# Disable unavailable channels in promote menu

Fixes #1204 
Fixes #1332

Disables unavailable channels in promote menu instead of hiding them.
Also, works around #1332 by using default contextual menu direction.

### QA
- ./run or demo
- go to Releases page of any snap
- try to promote any channel with devmode snaps in it
- menu should show stable and candidate channels, but they should be disabled
- open close menu of 'latest/candidate' - menu text should not be truncated
- if given channel can't be released to any other channel (for example beta with devmode snap) - menu button should be disabled (and not clickable)

<img width="805" alt="screen shot 2018-11-22 at 12 40 14" src="https://user-images.githubusercontent.com/83575/48900839-cb947300-ee53-11e8-8fda-7dee840fc559.png">
